kustomize/commands/internal/kustfile/kustomizationfile_test.go

```diff
@@ -386,3 +386,87 @@ foo:
 		t.Fatalf("Expect an unknown field error but got: %v", err)
 	}
 }
+
+func TestPreserveTrailingCommentsEndLine(t *testing.T) {
```

Comment:
nit: this and the following test cases could be combined in a test case table style.